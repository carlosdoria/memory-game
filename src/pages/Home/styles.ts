import styled from 'styled-components'

export const Main = styled.main`
  color: purple;
  h1 {
    display: flex;
    justify-content: center;
    font-size: 3rem;
    color: blue;
  }
`
