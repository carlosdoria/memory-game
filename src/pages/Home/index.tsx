import * as S from './styles'
import Link from 'next/link'

const Home = ( { title = 'Lista de jogos' } ) => {
  return (
    <S.Main>
      <h1>{title}</h1>
      <ol>
        <li >
          <Link href='MemoryGame'>
            Jogo da Mem√≥ria
          </Link>
        </li>
      </ol>
    </S.Main>
  )
}

export default Home
