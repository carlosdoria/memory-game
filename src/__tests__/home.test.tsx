import React from 'react';
import { screen, render } from '@testing-library/react';

import Home from '../pages';

describe('<Home />', () => {
  it('should render the heading', () => {
    const { container } = render(<Home />);

    expect(
      screen.getByRole('heading', { name: /my boilerplate/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the colors correctly', () => {
    const { container } = render(<Home />);

    expect(container.firstChild).toHaveStyle({ color: 'purple' });
  });
});
