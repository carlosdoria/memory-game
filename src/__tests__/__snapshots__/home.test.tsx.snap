// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the heading 1`] = `
.c0 {
  color: purple;
}

.c0 h1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 3rem;
  color: blue;
}

<main
  class="c0"
>
  <h1>
    My boilerplate
  </h1>
  <h2>
    First boilerplate
  </h2>
</main>
`;
